File: Models/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookPulse.Models
{
    public class Book
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(200)]
        public string Title { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string Author { get; set; } = string.Empty;

        [StringLength(50)]
        public string? ISBN { get; set; }

        [Required]
        public int TotalCopies { get; set; }

        [Required]
        public int AvailableCopies { get; set; }

        [Column(TypeName = "decimal(3,2)")]
        public decimal? AverageRating { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        public virtual ICollection<Loan> Loans { get; set; } = new List<Loan>();
        public virtual ICollection<Rating> Ratings { get; set; } = new List<Rating>();
    }
}

File: frontend/src/types/book.ts
export interface Book {
  id: number
  title: string
  author: string
  isbn?: string
  totalCopies: number
  availableCopies: number
  averageRating?: number
  createdAt: string
}

export interface CreateBookDto {
  title: string
  author: string
  isbn?: string
  totalCopies: number
}

export interface UpdateBookDto {
  title?: string
  author?: string
  isbn?: string
  totalCopies?: number
}
